py_bentoctl/state.py

```diff
@@ -0,0 +1,75 @@
+import json
+import sqlite3
+import sys
+
+from typing import Any, Dict, Optional
+
+from .config import BENTO_ORCHESTRATION_STATE_DB_FILE, BENTO_SERVICES_DATA
+
+__all__ = [
+    "MODE_DEV",
+    "MODE_PROD",
+    "get_db",
+    "get_state",
+    "set_state_services",
+]
+
+STATE_SCHEMA = """
+CREATE TABLE IF NOT EXISTS kvstore (
+    k TEXT PRIMARY KEY,
+    v TEXT
+);
+"""
+
+
+MODE_DEV = "dev"
+MODE_PROD = "prod"
+
+
+def get_db() -> sqlite3.Connection:
+    return sqlite3.connect(BENTO_ORCHESTRATION_STATE_DB_FILE)
+
+
+def get_state(conn: Optional[sqlite3.Connection] = None):
+    db = conn or get_db()
+
+    first_run_services = {
+        k: {"mode": "prod"}
```

Comment:
With the py scripts accepting .env values for MODE (dev or prod), I feel that enforcing prod mode no matter the env values here is going to confuse users if they have MODE=dev in local.env

Reply:
yeah thats probably true - i'll have to check what `MODE` does in `local.env` to see if we need it

I'm not sure if we really want to clone everything locally for dev purposes since most of the time most services aren't being worked on by a given dev